### PR TITLE
Replace Newtonsoft with System.Text.Json

### LINKED ISSUE #50: Consider migrating from Newtonsoft.Json to System.Text.Json
https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-5-0

### PR DESCRIPTION
**Pull request by adrianiftode**
Closes #50

### CODE REVIEW

File: src/FluentAssertions.Web/Internal/HttpContentExtensions.cs [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
-        public static async Task<T> ReadAsAsync<T>(this HttpContent content)
+        public static async Task<T?> ReadAsAsync<T>(this HttpContent content)
         {
-            var stringResponse = await content.ReadAsStringAsync();
-            return JsonConvert.DeserializeObject<T>(stringResponse);
+            var contentStream = await content.ReadAsStreamAsync();
+            return await JsonSerializer.DeserializeAsync<T>(contentStream, new JsonSerializerOptions
+            {
+                AllowTrailingCommas = true,
+                PropertyNameCaseInsensitive = true,
```
Comment (adrianiftode) [inline src/FluentAssertions.Web/Internal/HttpContentExtensions.cs:18]: remove trailing space, comma

File: src/FluentAssertions.Web/BadRequestAssertions.cs [hunk 2]
```diff
@@ -52,7 +52,7 @@ public BadRequestAssertions(HttpResponseMessage value) : base(value)
             Guard.ThrowIfArgumentIsNullOrEmpty(expectedErrorField, nameof(expectedErrorField), "Cannot verify having an error against a <null> or empty field name.");
             Guard.ThrowIfArgumentIsNullOrEmpty(expectedWildcardErrorMessage, nameof(expectedWildcardErrorMessage), "Cannot verify having an error against a <null> or empty wildcard error message.");
 
-            Func<Task<JToken>> jsonFunc = () => Subject.GetJsonObject();
+            Func<Task<JsonDocument>> jsonFunc = () => Subject.GetJsonDocument();
```
Comment (adrianiftode) [inline src/FluentAssertions.Web/BadRequestAssertions.cs:55]: Dispose the json, same for other usages

File: src/FluentAssertions.Web/Internal/JsonExtensions.cs [hunk 3]
```diff
@@ -1,68 +1,94 @@
...
+            var first = byKey.First();
+            switch (first.Value.ValueKind)
             {
-                JArray values => values.Select(c => (string)c),
-                { } token => new[] { token.ToString() },
-                _ => Enumerable.Empty<string>()
-            };
+                case JsonValueKind.Array:
+                    return first.Value.EnumerateArray().Select(c => c.ToString() ?? "");
+                default:
```
Comment (adrianiftode) [inline src/FluentAssertions.Web/Internal/JsonExtensions.cs:27]: Add tests to check the behavior for properties with complex values and with null, undefined values.

File: src/FluentAssertions.Web/Internal/JsonExtensions.cs [hunk 4]
```diff
@@ -1,68 +1,94 @@
...
-            json.AllTokens().Where(c => c.Type == JTokenType.Property
-                                        && string.Equals(((JProperty)c).Name, key, StringComparison.OrdinalIgnoreCase));
+        private static IEnumerable<(JsonProperty element, JsonProperty? parent)> GetByKeyWithParent(this JsonDocument json, string key)
+            => json.RootElement.AllPropertiesWithParent().Where(pair => string.Equals(pair.element.Name, key, StringComparison.OrdinalIgnoreCase));
 
-        private static IEnumerable<JToken> AllTokens(this JToken obj)
+        private static IEnumerable<JsonProperty> AllProperties(this JsonElement obj)
+            => obj.AllPropertiesWithParent().Select(pair => pair.element);
+
+        private static IEnumerable<(JsonProperty element, JsonProperty? parent)> AllPropertiesWithParent(this JsonElement obj)
```
Comment (adrianiftode) [inline src/FluentAssertions.Web/Internal/JsonExtensions.cs:56]: Rename element with property

File: src/FluentAssertions.Web/Internal/JsonExtensions.cs [hunk 5]
```diff
@@ -1,68 +1,94 @@
...
-                    toSearch.Push(child);
+                    foreach (var arrayElement in inspected.element.EnumerateArray())
+                    {
+                        if (arrayElement.ValueKind == JsonValueKind.Object)
+                        {
+                            foreach (var property in arrayElement.EnumerateObject())
+                            {
+                                toSearch.Push((property.Value, inspected.parent));
+                            }
+
```
Comment (adrianiftode) [inline src/FluentAssertions.Web/Internal/JsonExtensions.cs:75]: Remove extra line
